x-pack/auditbeat/module/system/user/_meta/data.json

```diff
@@ -0,0 +1,24 @@
+{
+    "@timestamp": "2017-10-12T08:05:34.853Z",
+    "beat": {
+        "hostname": "host.example.com",
+        "name": "host.example.com"
+    },
+    "event": {
+        "dataset": "user",
+        "module": "system"
+    },
+    "event.action": "existing_user",
+    "event.type": "state",
+    "system": {
+        "user": {
```

Comment:
But keep them under `system/user` as well? I.e.:

```
"user": {
  "uid": 1001,
  "gid": 1001,
  "name": "ubuntu"
},
"system": {
    "user": {
      "passwd": "x",
      "uid": 1001,
      "gid": 1001,
      "dir": "/home/ubuntu",
      "shell": "/bin/bash",
      "user_information": "Ubuntu",
      "name": "ubuntu"
    }
  },
```

Reply:
@ruflin @webmat Would you rather have some of the fields duplicated to `user.*` or should we move/promote the ECS fields (no duplication)?

Reply:
Ok, should the final result look like this:

```
"user": {
  "id": 1001,
  "name": "ubuntu"
},
"system": {
    "user": {
      "passwd": "x",
      "gid": 1001,
      "dir": "/home/ubuntu",
      "shell": "/bin/bash",
      "user_information": "Ubuntu",
    }
  },
```

Only `user.id` and `user.name` currently seem to be in ECS. `uid`/`gid` are not. Personally, I would prefer to use `uid` since it's such a common term.

Reply:
Copying data over or aliasing are both fine by me, I don't have a strong opinion on this part.

I absolutely think it's important to fill the ECS fields as much as possible, however :-)

Regarding the field name, `user.id` would indeed be the place to put the Linux uid at this time. I understand that in the context of Linux, uid and gid is the go to name for these values. However I don't see that as reason enough to define a second field under `user` (and eventually `group`). Other arbitrary systems (OS or otherwise) will rarely use this nomenclature, so this would give us a situation where almost all systems fill `user.id` and exactly one system -- Linux -- fills `user.uid`.

I'm totally ok if you use uid/gid under the `system_audit` namespace, however.

Reply:
To me at least the meaning of `user.id` would not obvious. If I'm a *NIX admin, I would wonder whether this is a custom ID generated by  Elastic Stack (for example, to identify the same user across systems where on each system they might have a different `uid`) rather than my well-known `uid`. As a Windows admin, I would wonder whether this is my `SID` or not. In any case, it's not obvious because it doesn't follow convention.

So I'd want to keep `system.user.uid` (and probably have a `system.user.sid` in the future for Windows). We can copy it to a top-level `user.id` (if we don't want to use that for some sort of cross-system identification).

Reply:
Sounds good. We now have a top-level `user.id` and a `system.user.uid` field that share the same value: https://github.com/elastic/beats/blob/29db6420babaa534416c968350103acaeb6ac01b/x-pack/auditbeat/module/system/user/_meta/data.json